commit 6ae1b4238c5c3851e7aabf80f9b122870e339c7a
Author: Edwin Török <edvin.torok@citrix.com>
Date:   Thu Oct 11 16:24:56 2018 +0100

    Revert "CP-28951: Call xapi to send message when receive xen lowmem event"
    
    This reverts commit dff4114e04faf50f50f65dadaecfce1f494810c9.

diff --git a/xc/jbuild b/xc/jbuild
index b76a63dc..bfe93df3 100644
--- a/xc/jbuild
+++ b/xc/jbuild
@@ -57,7 +57,6 @@ let () = Printf.ksprintf Jbuild_plugin.V1.send {|
     ppx_sexp_conv.runtime-lib
     re
     re.pcre
-    xeneventchn
   ))
 ))
 
diff --git a/xc/xenops_server_xen.ml b/xc/xenops_server_xen.ml
index 87182542..bbfdd539 100644
--- a/xc/xenops_server_xen.ml
+++ b/xc/xenops_server_xen.ml
@@ -3493,26 +3493,6 @@ let look_for_xenctrl () =
       exit 1;
     end
 
-let watch_xen_event () =
-  let open Xeneventchn in
-  let handle = init () in
-  let virq_enomem_port = bind_virq handle Enomem in
-  let host_uuid = Inventory.lookup Inventory._installation_uuid in
-  let rec process_event () =
-    debug "waiting for xen event at (%s)" __LOC__;
-    let port = pending handle in
-    if port = virq_enomem_port then begin
-      info "Sending HOST_LOW_MEMORY message.";
-      (* Call the script to call xenapi to create message which indicates the host is in low memory situation *)
-      let pid = Forkhelpers.safe_close_and_exec None None None [] !Xc_resources.send_message_script ["--name"; "HOST_LOW_MEMORY"; "--priority"; "2"; "--class"; "host"; "--uuid"; host_uuid; "--body"; "Host does not have enough memory in xen heap"]  in
-      Forkhelpers.dontwaitpid pid
-    end;
-    (* We want to receive further notification, so unmask it *)
-    if to_int port <> -1 then unmask handle port;
-    process_event ()
-  in
-  process_event ()
-
 let init () =
   look_for_forkexec ();
 
@@ -3548,17 +3528,6 @@ let init () =
   Device.Backend.init();
   debug "xenstore is responding to requests";
   let () = Watcher.create_watcher_thread () in
-  let (_: Thread.t) = Thread.create (fun () ->
-      debug "starting xen event watch thread.";
-      while true do
-        try
-          watch_xen_event ()
-        with e ->
-          debug "watch_xen_event thread raised: %s" (Printexc.to_string e);
-          debug "watch_xen_event thread backtrace: %s" (Printexc.get_backtrace ());
-          Thread.delay 5.;
-      done) ()
-  in
   ()
 
 module DEBUG = struct
